src/main/scala/com/microsoft/hyperspace/index/plananalysis/PlanAnalyzer.scala

```diff
@@ -212,7 +213,13 @@ object PlanAnalyzer {
       plan: SparkPlan,
       indexes: DataFrame,
       bufferStream: BufferStream): Unit = {
-    val usedIndexes = indexes.filter(indexes("indexLocation").isin(getPaths(plan): _*))
+    val usedIndexNames = plan.collect {
```

Comment:
Seems this will fix #432. 👍  It might be good to create a new PR for this.

Reply:
I'll check if it fixes #432 and create a PR containing tests.